symfit/core/minimizers.py

```diff
@@ -293,13 +332,28 @@ def execute(self, bounds=None, jacobian=None, constraints=None, **minimize_optio
             :func:`scipy.optimize.minimize`. Note that your `method` will
             usually be filled by a specific subclass.
         """
+        if bounds is None and isinstance(self, BoundedMinimizer):
```

Comment:
Either way, I think this is getting out of scope for this PR.

Reply:
I think the idea of having one deal with init, and the other with execute, is a nice idea. I just don't know if it will scale, but for now that would work for all our objects.

Between those two options, I think option 1 is the cleanest. The inheritance could also be drastically cleanup by having methods inherit from each other, which is code technically the right thing to do although conceptually it is weird.

For example, 
```python
class BFGS(GradientMinimize, ScipyGradientMinimize):
    pass

class LBFGSB(BFGS, BoundedMinimize, ScipyBoundedMinimize):
    pass

class SLSQP(LBFGSB, ConstrainedMinimize, ScipyConstrainedMinimize):
    pass
```
Codewise this is a lot cleaner although it is not true that SLSQP inherits from LBFGSB as an algorithm.

Looking at this code now though, it does seem a bit silly to type things twice. Maybe we can make a seperate PR to dive into the MRO again, there has to be a better way.

Reply:
This is not something to rush into indeed. I'd say this is out of scope for this PR. We should let it stew for a while, and revisit the minimizer MRO/inheritance tree again later.
Let's leave this as is for now (it works, yay).